_data/rules.yml

```diff
@@ -223,6 +223,10 @@ groups:
                 description: '{{ $labels.instance }} interface {{ $labels.device }} has encountered {{ printf "%.0f" $value }} transmit errors in the last five minutes.'
                 query: 'increase(node_network_transmit_errs_total[5m]) > 0'
                 severity: warning
+              - name: Host Network Not 1Gb Speed
+                description: 'The network interface "{{ $labels.device }}" on "{{ $labels.instance }}" is not connected at 1Gb.'
+                query: 'node_network_speed_bytes != 125000000'
```

Comment:
It looks quite specific to your environment and not actionable. I can say it will definitely fire in mine env - mixed 100Mb,1Gb,10Gb, so why you want to promote this as a generic alert?

If you want to detect problems then probably checking if port speed changed over time would be a better approach then checking if it is equal to some magic number.

Reply:
If it’s the wrong speed it’s actionable. 

But yes, this would be specific to a 1Gb environment. You would have to tailor it if required. Is that not expected?

Reply:
Correct me if i'm wrong, but i think this metric is expected to be stable, like the total memory on your system.

Something like `irate(node_network_receive_bytes_total[1m]) / node_network_speed_bytes > 0.8` (80%) seems more appropriate here.

It does not monitor exactly the same thing, but it's more accurate, because what you want, at the end of the day, is ensuring you have enough bandwidth.

WDYT ?

Reply:
I agree with @samber  I personnaly use this alert that ensure total bandwidth do not exceed the iface's capacity
```
(
            rate(node_network_receive_bytes_total{device!~"^tap.*"}[1m])
            +
            rate(node_network_transmit_bytes_total{device!~"^tap.*"}[1m])
          )
          /
          node_network_speed_bytes{device!~"^tap.*"} > 0.8
```
(I skip tap device become tap driver does not "support" maximal speed)

Reply:
@Turgon37 I like your query 👍